### PR TITLE
add greenfield authorize UI docs and fix small issues

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Don't use .ToString, but AbsoluteUri

**Comment by NicolasDorier** [comment]
https://faithlife.codes/blog/2010/08/uritostring_must_die/

File: BTCPayServer/Controllers/ManageController.APIKeys.cs [hunk 1]
```diff
@@ -266,7 +271,7 @@ public async Task<IActionResult> AuthorizeAPIKey([FromForm] AuthorizeApiKeysView
                 case "authorize":
                 case "confirm":
                     var key = command == "authorize"
-                        ? await CreateKey(viewModel, (viewModel.ApplicationIdentifier, viewModel.RedirectUrl?.Authority))
+                        ? await CreateKey(viewModel, (viewModel.ApplicationIdentifier, viewModel.RedirectUrl.ToString()))
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/ManageController.APIKeys.cs:274]: `.AbsoluteUri`

File: docs/greenfield-authorization.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+
+# GreenField API Authorization Flow
+
+The GreenField API allows two modes of authentication to its endpoints: Basic auth and API Keys. 
+
+## Basic auth
+Basic auth allows you to seamlessly integrate with BTCPay Server's user system using only a traditional user/password login form. This is however a security risk if the application is a third party as they will receive your credentials in plain text and will be able to access your full account.
+
+## API Keys
+BTCPay Server's GreenField api also allows users to generate API keys with [specific permissions](https://docs.btcpayserver.org/API/Greenfield/v1/#section/Authentication/API%20Key). **If you are integrating BTCPay Server into your third party application, this is the recommended way.**
```
Comment (dennisreimann) [inline docs/greenfield-authorization.md:10]:
```suggestion
BTCPay Server's GreenField API also allows users to generate API keys with [specific permissions](https://docs.btcpayserver.org/API/Greenfield/v1/#section/Authentication/API%20Key). **If you are integrating BTCPay Server into your third-party application, this is the recommended way.**
```

File: docs/greenfield-authorization.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Asking a user to generate a dedicated API key, with a specific set of permissions can be a bad UX experience. For this scenario, we have the [Authorize User UI](https://docs.btcpayserver.org/API/Greenfield/v1/#tag/Authorization). This allows external applications to request the user to generate an API key with a specific set of permissions by simply generating a url to BTCPay Server and redirecting the user to it.
```
Comment (dennisreimann) [inline docs/greenfield-authorization.md:12]:
```suggestion
Asking a user to generate a dedicated API key, with a specific set of permissions can be a bad UX experience. For this scenario, we have the [Authorize User UI](https://docs.btcpayserver.org/API/Greenfield/v1/#tag/Authorization). This allows external applications to request the user to generate an API key with a specific set of permissions by simply generating a URL to BTCPay Server and redirecting the user to it.
```

File: docs/greenfield-authorization.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Additionally, there are 2 optional parameters to the endpoint which allow a more seamless integration:
+* if `redirect` is specified, once the API key is created, BTCPay server redirects the user via a POST submission to the specified `redirect` url, with a json body containing the api key, user id, and permissions granted.
```
Comment (dennisreimann) [inline docs/greenfield-authorization.md:14]:
```suggestion
* if `redirect` is specified, once the API key is created, BTCPay Server redirects the user via a POST submission to the specified `redirect` URL, with a json body containing the API key, user id, and permissions granted.
```

File: docs/greenfield-authorization.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+* if `applicationIdentifier` is specified (along with `redirect`), BTCPay Server will check if there is an existing API key associated with the user that also has this application identifier, redirect host AND the permissions required match. `applicationIdentifier` is ignored if `redirect` is not specified.
+
+Some examples of a generated Authorize url:
```
Comment (dennisreimann) [inline docs/greenfield-authorization.md:17]:
```suggestion
Some examples of a generated Authorize URL:
```

File: docs/greenfield-authorization.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+* https://mainnet.demo.btcpayserver.org/api-keys/authorize?applicationName=Your%20Application&redirect=http://gozo.com&applicationIdentifier=gozo - Attempts to match a previously created API key based on the app identifier, domain and permissions and is prompted.
+* https://mainnet.demo.btcpayserver.org/api-keys/authorize?permissions=btcpay.store.cancreateinvoice&permissions=btcpay.store.canviewinvoices - A request asking for permissions to create and view invoices on all stores available to the user
+* https://mainnet.demo.btcpayserver.org/api-keys/authorize?permissions=btcpay.store.cancreateinvoice&permissions=btcpay.store.canviewinvoices&selectiveStores=true - A request asking for permissions to create and view invoices on stores but also allows the user to choose which stores the application will have the permission to.
+* https://mainnet.demo.btcpayserver.org/api-keys/authorize?permissions=btcpay.store.cancreateinvoice&permissions=btcpay.store.canviewinvoices&strict=false - A request asking for permissions but allows the user to remove or add to the requested permission list. 
+
+
+
```
Comment (dennisreimann) [inline docs/greenfield-authorization.md:27]:
```suggestion
* `https://mainnet.demo.btcpayserver.org/api-keys/authorize` - A simplistic request, where no permission is requested. Useful to prove that a user exists on a specific BTCPay Server instance. 
* `https://mainnet.demo.btcpayserver.org/api-keys/authorize?applicationName=Your%20Application` - Indicates that the API key is being generated for `Your Application`
* `https://mainnet.demo.btcpayserver.org/api-keys/authorize?applicationName=Your%20Application&redirect=http://gozo.com` - Redirects the user via a POST to `http://gozo.com` with a JSON body containing the API key and its info.
* `https://mainnet.demo.btcpayserver.org/api-keys/authorize?applicationName=Your%20Application&redirect=http://gozo.com&applicationIdentifier=gozo` - Attempts to match a previously created API key based on the app identifier, domain and permissions and is prompted.
* `https://mainnet.demo.btcpayserver.org/api-keys/authorize?permissions=btcpay.store.cancreateinvoice&permissions=btcpay.store.canviewinvoices` - A request asking for permissions to create and view invoices on all stores available to the user
* `https://mainnet.demo.btcpayserver.org/api-keys/authorize?permissions=btcpay.store.cancreateinvoice&permissions=btcpay.store.canviewinvoices&selectiveStores=true` - A request asking for permissions to create and view invoices on stores but also allows the user to choose which stores the application will have the permission to.
* `https://mainnet.demo.btcpayserver.org/api-keys/authorize?permissions=btcpay.store.cancreateinvoice&permissions=btcpay.store.canviewinvoices&strict=false` - A request asking for permissions but allows the user to remove or add to the requested permission list. 

```

**Review by dennisreimann — COMMENTED** [review]
Just some minor improvements, wording consistency and typo fixes.

**Pull request merged**